lib/types/facebook.js

```diff
@@ -1,14 +1,44 @@
 const type = 'facebook';
+import {element} from 'deku';
+import parseText from '../parse-text';
+import renderText from '../render-text';
+import last from 'lodash.last';
+
+const parsePost = elm => {
+  const url = elm.getAttribute('data-href');
+  const embedAs = elm.classList.contains('fb-video') ? 'video' : 'post';
+  const pElm = elm.getElementsByTagName('p')[0];
+  const text = parseText(pElm);
+  const aElms = elm.getElementsByTagName('a') || [];
+  const user = aElms[0] ? aElms[0].childNodes[0].data : '';
+  const date = aElms[1] ? aElms[1].childNodes[0].data : '';
+
+  return { embedAs, type, url, text, date, user };
+};
+
+const parseVideo = elm => {
+  const url = elm.getAttribute('data-href');
+  const embedAs = elm.classList.contains('fb-video') ? 'video' : 'post';
+  const aElms = elm.getElementsByTagName('a') || [];
+  const headline = aElms[0] ? aElms[0].childNodes[0].data : '';
```

Comment:
I use `[0]` since it's more secure.

So perhaps it's not valid in this case, but we could have `aElms = [null]` and then checking length wouldn't be enough. That's why I do `aElms[0]`.